geomstats/geometry/product_manifold.py

```diff
@@ -321,7 +320,7 @@ def _pool_outputs_from_function(self, outputs):
             and gs.all([gs.is_bool(factor_output) for factor_output in outputs])
             or (not all_arrays)
         ):
-            outputs = gs.stack(outputs)
+            outputs = gs.stack([gs.array(factor_output) for factor_output in outputs])
```

Comment:
do you remember what's the case that messed up with this line before you changes?

Reply:
Unfortunately no, it was quite a long time ago...